content/docs/user-guide/gems/reference/aws/aws-gamelift/flexmatch/cpp-api.md

```diff
@@ -0,0 +1,257 @@
+---
+linkTitle: FlexMatch C++ API
+title: FlexMatch C++ API 
+description: Learn how to use the C++ API for the AWS GameLift Gem in Open 3D Engine (O3DE).
+toc: true
+weight: 300
+---
+
+
+## Client Side
+
+The **AWS GameLift** Gem implements a matchmaking interface for Amazon GameLift. The *matchmaking interface* (`IMatchmakingRequests` and `IMatchmakingAsyncRequests`) provides public APIs to add player matchmaking functionality to your GameLift hosted games. 
+
+The matchmaking interface represents the actions required to handle matchmaking in a game. The Gem provides an implementation to this interface which utilizes GameLift FlexMatch to find and group players in a session for your game.
+
+There must be only one implementation of the matchmaking interface per matchmaking solution. To add support for another matchmaking solution, you must create another implementation of the matchmaking interface.
+
+
+### Matchmaking APIs
+
+#### `StartMatchmaking`
```

Comment:
Do we want to keep the line above (### Matchmaking APIs) as H3 as well? That seems a little weird since `StartMatchmaking` should be one of the sections under Matchmaking APIs.

Reply:
I wrestled with different ideas on how to format this section to include all the relevant information categorized under headings _and_ show all the APIs in the righthand ToC. I tried a few different ways, and while it might seem weird at first to us authors to have both "Machmaking APIs" and each following API at the same H3 level, I think the reader will still find it intuitive when it publishes because of the different fonts that will be used, _and_ they'll find it useful to have all the APIs included in the page ToC. (H4 level is not included in this ToC.)